### PR TITLE
feat: Upgrade OpenDAL to v0.16.0

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will bump OpenDAL to v0.16.0

- https://github.com/datafuselabs/opendal/discussions/631
- https://github.com/datafuselabs/opendal/discussions/660

### CODE REVIEW

**Comment by BohuTANG** [comment]
```
error[E0599]: no method named `status` found for enum `std::option::Option` in the current scope
   --> /opt/rust/cargo/registry/src/github.com-1ecc6299db9ec823/http-types-2.12.0/src/error.rs:113:70
    |
113 |         if let std::backtrace::BacktraceStatus::Captured = backtrace.status() {
    |                                                                      ^^^^^^ method not found in `std::option::Option<&Backtrace>`
    |
   ::: /opt/rust/cargo/registry/src/github.com-1ecc6299db9ec823/http-types-2.12.0/src/status.rs:11:8
```

**Comment by Xuanwo** [comment]
Yep, I'm working on this.

**Comment by Xuanwo** [comment]
To address the backtrace issue, I bumped `openraft` to v0.7.2.

Cc @drmingdrmer for confirmation.

File: src/meta/types/Cargo.toml [hunk 1]
```diff
@@ -15,10 +15,10 @@ common-datavalues = { path = "../../query/datavalues" }
 common-exception = { path = "../../common/exception" }
 common-storage = { path = "../../common/storage" }
 
-openraft = { git = "https://github.com/drmingdrmer/openraft", rev = "v0.7.2-alpha.4" }
+openraft = "0.7.2"
```
Comment (drmingdrmer) [inline src/meta/types/Cargo.toml:18]: What about keeping using the git source?

I'm afraid the next update will still be alpha, and it has to revert to using git source. 🤔

**Review by drmingdrmer — APPROVED** [review]

File: src/meta/types/Cargo.toml (hunk 1 above)
Comment (Xuanwo) [reply src/meta/types/Cargo.toml]: > What about keeping using the git source?

Addressed. We are tracking the latest commit on `github.com/datafuselabs/openraft` branch `release-0.7`.

**Comment by Xuanwo** [comment]
This PR also cleanup the blocks of #7466

**Review by PsiACE — APPROVED** [review]

File: src/binaries/Cargo.toml [hunk 2]
```diff
@@ -47,8 +47,9 @@ databend-meta = { path = "../meta/service" }
 databend-query = { path = "../query/service" }
 
 # Crates.io dependencies
-anyhow = "1.0.58"
+anyhow = "1.0.64"
 clap = { version = "3.2.5", features = ["derive", "env"] }
+openraft = { git = "https://github.com/datafuselabs/openraft", rev = "6057abbf" }
```
Comment (drmingdrmer) [inline src/binaries/Cargo.toml:52]: A `rev` is unstable: a commit can be removed by the author.

The tag won't be removed(by convention). And a tag name is also more clear for developers.

Comment (Xuanwo) [reply src/binaries/Cargo.toml]: Addressed

**Review by drmingdrmer — APPROVED** [review]

**Comment by Xuanwo** [comment]
The test case for HTTP location failed. I'm working on it.

**Comment by Xuanwo** [comment]
Related bug: https://github.com/datafuselabs/opendal/issues/670

**Comment by Xuanwo** [comment]
> test scalars::string::test_string ... FAILED

Seems not related to this PR.

Cc @andylokandy for confirmation.

**Comment by andylokandy** [comment]
It should be fixed via https://github.com/datafuselabs/databend/pull/7583

**Comment by Xuanwo** [comment]
> It should be fixed via #7583

This PR needs to be merged first as another PRs depends on this. So I give a quick workaround.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**